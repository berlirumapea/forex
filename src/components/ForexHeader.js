import React, { Component } from "react";
import { Card } from "semantic-ui-react";

export default class ForexHeader extends Component {

	state = {
		val: 10.00
	}

	onChange = (e) => this.setState({ val: e.target.value});

  render() {
    return (
      <Card.Content>
        <Card.Header>
          <p id="title">USD - United State Dollars</p>
          <div className="flex-between">
            <h3 className="bold">USD</h3>
            <input
              value={this.state.val}
              type="number"
              step="0.01"
              onChange={this.onChange}
							autoFocus
							className="usd"
            />
          </div>
        </Card.Header>
      </Card.Content>
    );
  }
}
