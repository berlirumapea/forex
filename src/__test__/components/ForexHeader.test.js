import React from "react";
import ForexHeader from "../../components/ForexHeader";
import { shallow } from "enzyme";

describe("<ForexHeader /> static data testing", () => {
  it("renders a paragraph element and has value of USD - United State Dollars", () => {
    const wrapper = shallow(<ForexHeader />);
    const p = wrapper.find("#title");
    expect(p).toBeDefined();
    expect(p.text()).toMatchSnapshot();
  });

  it("renders h3 element and has value of USD", () => {
    const wrapper = shallow(<ForexHeader />);
    const h3 = wrapper.find("h3");
    expect(h3).toBeDefined();
    expect(h3.text()).toMatchSnapshot();
  });

  it("has a input element with default value of 10.00", () => {
    const wrapper = shallow(<ForexHeader />);
    const input = wrapper.find("input");
    expect(input.props().value).toEqual(10.0);
  });

  it("renders a div with className == flex-between that wraps input element", () => {
    const wrapper = shallow(<ForexHeader />);
    let div = wrapper.find(".flex-between");
    let input = div.find("input.usd");
    expect(div).toBeDefined();
    expect(input).toBeDefined();
  });
});

describe("<ForexHeader /> user interactions testing", () => {
  it("updates the state when input has changed", () => {
    const wrapper = shallow(<ForexHeader />);
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: 20 } });
		expect(wrapper.state("val")).toEqual(20);
		input.simulate("change", { target: { value: 40 } });
		expect(wrapper.state("val")).toEqual(40);
  });
});
