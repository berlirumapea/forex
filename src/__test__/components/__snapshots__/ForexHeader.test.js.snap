// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForexHeader /> static data testing renders a paragraph element and has value of USD - United State Dollars 1`] = `"USD - United State Dollars"`;

exports[`<ForexHeader /> static data testing renders h3 element and has value of USD 1`] = `"USD"`;
